### DISCUSSION TITLE
How do I parse my own encapsulated response method?

### DISCUSSION THREAD

**Discussion by terranc** [dedoc/scramble#148] [Q&A] [answered]
The code is as follows:
Controller:
```php
// ...

    public function success($data, $msg = ''): \Illuminate\Http\JsonResponse
    {
        return response()->json([
            'code' => 1,
            'msg'  => $msg,
            'data' => $data,
        ]);
    }
    public function create() {
        // ...
        return $this->success(new CreateBookingResource(...))
    }

// ... 
```

It seems that such a situation cannot be interpreted correctly.
What is the right way to do it?

**Comment by palonza** [comment]


**Reply by terranc** [reply]
@palonza Scramble\Scramble this namespace does not exist.
Also, try to introduce a Dedoc\Scramble\Scramble, found by no encode method

**Comment by romalytvynenko** [comment, marked as answer]
@terranc Hey

Currently there is no way to do it besides explicitly returning `response()->json(...)` from `create`. 

But this code pattern will be automatically documentable starting from version 0.8.0!
